specifyweb/frontend/js_src/lib/components/QueryBuilder/Header.tsx

```diff
@@ -88,6 +90,17 @@ export function QueryHeader({
           </ProtectedTable>
         </ProtectedAction>
       )}
+      {hasToolPermission(
+        'queryBuilder',
+        queryResource.isNew() ? 'create' : 'update'
+      ) && (
```

Comment:
The button should be visible to all users, regardless of permissions, right? Thus, should remove this check for permissions